src/cluster_asm.c

```diff
@@ -723,6 +699,33 @@ void asmMigrateFailed(asmTask *task) {
     clusterAsmOnEvent(task->slot_ranges, ASM_EVENT_MIGRATE_FAILED, NULL);
 }
 
+void asmTaskSetFailed(asmTask *task, const char *fmt, ...) {
```

Comment:
Single function to mark task failed. It logs and saves the error. So, we can avoid doing it separately where we call this function.

Reply:
LGTM